cpp/arcticdb/processing/operation_dispatch_ternary.cpp

```diff
@@ -0,0 +1,582 @@
+// lazy_df = lazy_df[where(lazy_df["col1"] < 0, True, False)]
+// although this is dumb, as the same effect could be achieved with:
+// lazy_df = lazy_df[lazy_df["col1"] < ]
+// It is actually here to cope with cases where input processing produced FullResult or EmptyResult, which are then
+// combined with a fixed bool value
+VariantData ternary_operator(const util::BitSet& condition, bool left, bool right) {
+    util::BitSet output_bitset;
+    if (left && right) {
+        if (condition.size() > 0) {
+            output_bitset.set_range(0, condition.size() - 1);
+        }
+    } else if (left) {
+        // right is false
+        output_bitset = condition;
+    } else if (right) {
+        // left is false
+        output_bitset = ~condition;
+    } // else left and right are false, the resize below will correctly initialise all bits to zero
+    output_bitset.resize(condition.size());
+    return output_bitset;
+}
+
+VariantData visit_ternary_operator(const VariantData& condition, const VariantData& left, const VariantData& right) {
+    if (std::holds_alternative<FullResult>(condition)) {
+        return left;
+    } else if (std::holds_alternative<EmptyResult>(condition)) {
+        return right;
+    }
+    auto transformed_condition = transform_to_bitset(condition);
+    // transformed_condition is a bitset. transform_to_bitset throws if provided a Value or ValueSet. If it is a column,
+    // it throws if the column type is not bool, and converts bool columns to a bitset, and full/empty results were
+    // handled above
+    auto c = std::get<util::BitSet>(std::move(transformed_condition));
+    // Of the possible types in VariantData, all except ValueSet can make sense here. Using the ordering:
+    // BitSet
+    // ColumnWithStrings
+    // Value
+    // FullResult
+    // EmptyResult
+    // All such combinations, including the reversing of arguments, are handled below
+    return std::visit(util::overload{
+            [&c](const util::BitSet &l, const util::BitSet &r) -> VariantData {
+                auto result = ternary_operator(c, l, r);
+                return transform_to_placeholder(result);
+            },
+            [&c](const util::BitSet &l, const ColumnWithStrings &r) -> VariantData {
+                auto bitset = std::get<util::BitSet>(transform_to_bitset(r));
+                auto result = ternary_operator(c, l, bitset);
+                return transform_to_placeholder(result);
+            },
+            [&c](const ColumnWithStrings &l, const util::BitSet &r) -> VariantData {
+                auto bitset = std::get<util::BitSet>(transform_to_bitset(l));
+                auto result = ternary_operator(c, bitset, r);
+                return transform_to_placeholder(result);
+            },
+            [&c](const util::BitSet &l, const std::shared_ptr<Value> &r) -> VariantData {
+                user_input::check<ErrorCode::E_INVALID_USER_ARGUMENT>(is_bool_type(r->data_type_),
```

Comment:
IIUC this is because the operator needs to resolve to a filter - might be worth explaining that in the error message or in a comment as the reason for the constraint wasn't obvious to me till I stepped through the tests